src/MultiRegion/multi_region_grid.jl

```diff
@@ -184,6 +184,7 @@ end
 function reconstruct_global_grid(mrg::ImmersedMultiRegionGrid) 
     global_grid     = reconstruct_global_grid(mrg.underlying_grid)
     global_boundary = reconstruct_global_boundary(mrg.immersed_boundary)
+    global_boundary = on_architecture(architecture(mrg), global_boundary)
```

Comment:
```suggestion
    global_ib = reconstruct_global_boundary(mrg.immersed_boundary)
    global_ib = on_architecture(architecture(mrg), global_ib)
```

when we leave the word "immersed" from "immersed boundary" things quickly get confusing